logger-cli/src/logger.rs

```diff
@@ -0,0 +1,156 @@
+// shared-core - bitdrift's common client/server libraries
+// Copyright Bitdrift, Inc. All rights reserved.
+//
+// Use of this source code is governed by a source available license that can be found in the
+// LICENSE file or at:
+// https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt
+
+use crate::cli::{FieldPairs, LogCommand, Options};
+use crate::metadata::Metadata;
+use crate::storage::SQLiteStorage;
+use bd_client_common::error::handle_unexpected;
+use bd_logger::{CaptureSession, InitParams, Logger};
+use bd_session::{Strategy, fixed};
+use bd_test_helpers::metadata_provider::LogMetadata;
+use parking_lot::Mutex;
+use std::path::{Path, PathBuf};
+use std::pin::Pin;
+use std::sync::Arc;
+
+pub type LoggerFuture =
+  Pin<Box<dyn Future<Output = anyhow::Result<()>> + 'static + std::marker::Send>>;
+
+pub const SESSION_FILE: &str = "session_id";
+
+#[allow(dead_code)]
+pub struct LoggerHolder {
+  logger: Logger,
+  future: Mutex<Option<LoggerFuture>>,
+  shutdown_trigger: bd_shutdown::ComponentShutdownTrigger,
+}
+
+impl LoggerHolder {
+  pub fn new(
+    logger: Logger,
+    future: LoggerFuture,
+    shutdown_trigger: bd_shutdown::ComponentShutdownTrigger,
+  ) -> Self {
+    Self {
+      logger,
+      future: Mutex::new(Some(future)),
+      shutdown_trigger,
+    }
+  }
+
+  pub fn start(&self) {
+    let Some(future) = self.future.lock().take() else {
+      return;
+    };
+
+    sleep(2000); // some initialization time for the network
+
+    std::thread::spawn(move || {
+      tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()
+        .unwrap()
+        .block_on(async {
+          handle_unexpected(future.await, "logger top level run loop");
```

Comment:
not sure if handle_unexpected will do anything if we haven't configured an error reporter so this can probably just unwrap or something?